File: 34. Find First and Last Position of Element in Sorted Array/Solution.java
import javax.lang.model.element.Element;

public class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] range = new int[]{-1, -1};
        int ptr = 0;
        boolean started = false;


        while (ptr < nums.length) {
            if (nums[ptr] == target && started == false) {
                range[0] = ptr;
                range[1] = ptr;
                ptr += 1;
                started = true;
            } else if (nums[ptr] != target && started == true) {
                range[1] = ptr - 1;
                ptr += 1;
                break;
            } else if (nums[ptr] == target && started == true) {
                range[1] = ptr;
                ptr += 1;
            } else {
                ptr += 1;
            }
        }

        return range;
    }
}

File: 34. Find First and Last Position of Element in Sorted Array/Solution.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        if (!nums.size()) return {-1, -1};
        int l = 0, r = nums.size() - 1;
        int start = -1, end = -1;
        while (l < r)
        {
            int mid = l + r >> 1;
            if (nums[mid] >= target) r = mid;
            else l = mid + 1;
        }
        if (nums[l] == target) start = l;
        
        r = nums.size() - 1;
        while (l < r)
        {
            int mid = l + r + 1 >> 1;
            if (nums[mid] <= target) l = mid;
            else r = mid - 1;
        }
        if (nums[l] == target) end = l;
        
        return {start, end};
    }
};